docs/developer-guide/syntax-reference.rst

```diff
@@ -959,6 +959,27 @@ Example:
       FROM users
       WHERE user_id LIKE 'santa%';
 
+BETWEEN
+~~~~~~~
+
+**Synopsis**
+
+.. code:: sql
+
+    WHERE expression [NOT] BETWEEN start_expression AND end_expression;
+
+The BETWEEN operator is used to indicate that a certain value must lie within
```

Comment:
> Currently, KSQL supports numeric and string values for comparison.

Probably worth fleshing this out a bit. eg. arithmetic is supported, so is struct, array, map access. Anything else? Function calls aren't, anything else?  This probably also highlight any missing JSON tests, both negative and positive.

Reply:
Totally agree. I will rephrase this to: "Currently, KSQL supports any expression that resolves to a numeric or string value for comparison". Does that cover things?